### DISCUSSION TITLE
[Endpoints] Unexpected behaviour of api-gateway on request with secret

### DISCUSSION THREAD

**Discussion by SalvadorCali** [mia-platform/community#157] [Q&A] [answered]
Hi, everyone!
We added a **cross-project proxy** endpoint on our gateway project (we'll call it PROJECT A) with the following configuration:

![image](https://user-images.githubusercontent.com/37811007/210403989-e2121b1b-ab04-4472-bcbd-af0a5ccaa616.png)

then, on the project where is deployed the service with the requested API (we'll call it PROJECT B), we added another endpoint with the following configuration, that is reached by the previous one:

![image](https://user-images.githubusercontent.com/37811007/210404496-90d6d601-8147-42a6-928d-7ebc69531e35.png)

We noticed that if we try to call the API, it reaches PROJECT B's `authorization-service` that returns 200, but the `api-gateway` returns `UNAUTHORIZED` because it finds a secret in the headers of the request, altough we specify that the API key is not required and the User Group Permission is set to true.
Do you know why it shows this behaviour?

**Comment by fredmaggiowski** [comment, marked as answer]
Is the client-key (or secret) header provided? If so the value must still be a valid value otherwise the request will be rejected. The fact that is not required does not mean that can be invalid

**Reply by SalvadorCali** [reply]
Thank you for your answer!
> Is the client-key (or secret) header provided?

Yes

> If so the value must still be a valid value otherwise the request will be rejected. The fact that is not required does not mean that can be invalid

Ok, but due to this behaviour I think that:
- it's not clear to understand that the request is rejected due to an invalid secret
- it seems counterintuitive that if I set an endpoint without any security check, it rejects the request due to an invalid secret

**Reply by fredmaggiowski** [reply]
The answer to both your points is that being the client-key used to understand if the request has the proper authorization material it is actually an intended behavior not to be explicit about what failed in the authorization process to prevent disclosing information to the outside world. 


>it's not clear to understand that the request is rejected due to an invalid secret

Logs from the authorization service should help you understand the issue, did they? 
Also if you are using the nginx api-gateway you could use the x-mia-debug header to understand what is going wrong in api resolution, have you tried it? 


> it seems counterintuitive that if I set an endpoint without any security check, it rejects the request due to an invalid secret

An invalid client-key makes an invalid request, and being an authorization material the request is rightfully rejected, the fact that "is not required" does not mean it can be invalid, only that it is ok for the user not to provide it

However [the same problem occurred a few days ago](https://github.com/mia-platform/community/discussions/131) so I guess documentation should be improved

**Reply by SalvadorCali** [reply]
Ok, clear, thank you!

**Reply by fredmaggiowski** [reply]
You're welcome :)
